### ISSUE TITLE
JG Router duplicate pages with more then 1 JG menu items

### ISSUE THREAD

**Issue by sergeytolkachyov** [JoomGalleryfriends/JoomGallery#36]
I have a 5 JoomGallery menu items:
- Parent category
- - Child category 1
- - Child category 2
- - Child category 3
- - Child category 4

JoomGallery categories structure is 
- Parent category
- - Child category 1
- - - SubChild category 1
- - - SubChild category 2
- - - SubChild category 3
- - - SubChild category 4
- - -  Many subchild categories
- - Child category 2
- - - SubChild category 1
- - - SubChild category 2
- - - SubChild category 3
- - - SubChild category 4
- - -  Many subchild categories
- - Child category 3
- - Child category 4

And have a menu Item aliases for this in other menu. So, when I go to **Parent category / Child category / Subchild category** I have true url. When I use com_search - I have an invalid url:  **Parent category / Subchild category** without "Child Category". Joomgallery router needs to have menu ItemId for true urls, but it not got it via com_search.

**Comment by MrMusic** [comment]
Yes, there can be a routing problem if there are several menu entries for the JoomGallery. 
Does the problem occur even if you only have only one menu entry of the menu type 'gallery view'?

**Comment by sergeytolkachyov** [comment]
> Yes, there can be a routing problem if there are several menu entries for the JoomGallery.
> Does the problem occur even if you only have only one menu entry of the menu type 'gallery view'?
I'll test it.

**Comment by sergeytolkachyov** [comment]
My first problem is search results from search module. It duplicate pages because it can't get a true  ItemId.
The problem is that JoomGallery use only one ItemId from JG settings or database. The function `getItemid` in  `components/com_joomgallery/helpers/ambit.php`. First time it tries to find `view=gallery` in __menu table. If it is not exists it tries to find menu items with `option=com_joomgallery` in link. Then this code https://github.com/JoomGalleryfriends/JoomGallery/blob/master/components/com_joomgallery/helpers/ambit.php#L436
```php
 $Itemid = $db->loadResult();
```
return only one - first ItemId. But there we need in more a more complex logic where more then only one JoomGallery menu item was found.

**Comment by sergeytolkachyov** [comment]
The same way is in https://github.com/JoomGalleryfriends/JoomGallery/blob/master/components/com_joomgallery/helpers/routing.php#L70
function **checkItemid**
There use only one menu ItemId.
